src/Cli/dotnet/commands/dotnet-test/IPC/NamedPipeBase.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable IDE0240 // Remove redundant nullable directive
+#nullable disable
+#pragma warning restore IDE0240 // Remove redundant nullable directive
+
+using System.Globalization;
+
+namespace Microsoft.DotNet.Tools.Test;
+
+internal abstract class NamedPipeBase
+{
+    private readonly Dictionary<Type, object> _typeSerializer = [];
+    private readonly Dictionary<int, object> _idSerializer = [];
+
+    public void RegisterSerializer<T>(INamedPipeSerializer namedPipeSerializer)
```

Comment:
In such context, I think it's better to have an argument of type `Type`. It's never possible to get the compiler infer the type because it's not used as argument.